vignettes/syntax_extractor.Rmd

```diff
@@ -0,0 +1,305 @@
+---
+title: "Syntax Extractor"
+output: 
+  html_document:
+    toc: true
+    number_sections: true
+vignette: >
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteIndexEntry{Kaiaulu Syntax Extractor}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r eval=FALSE}
+rm(list = ls())
+seed <- 1
+set.seed(seed)
+
+# Load libraries
+  require(kaiaulu)
+  require(data.table)
+  require(yaml)
+  require(stringi)
+  require(XML)
+  require(gt)
+```
+
+# Introduction
+
+In open-source projects, code is often spread across many files and organized in various ways. Understanding this code at a glance can be difficult, especially when projects grow large and complex. This is where syntax extraction comes in.
+
+The Syntax Extractor is a tool that helps us dive into the structure of source code by using an annotated format. It works by converting code into a structured XML format using srcML, making it easier to analyze and understand the underlying architecture of the project. By extracting key pieces of information such as class names, functions, or even comments (like documentation), we can generate meaningful data that helps with things like semantic analysis, code comprehension, and more.
+
+Imagine a codebase as a collection of books in a library. Without a catalog, finding the right book would be nearly impossible. Syntax extraction is like building a catalog for a codebase. It allows us to quickly pinpoint where classes, methods, and important comments are located, and organize this data in a structured way.
+
+For Kaiaulu, this is particularly useful as we move toward deeper analysis, such as applying machine learning techniques like word embeddings to the extracted syntax. This allows us to not only understand how the code is written but also how the pieces relate to each other on a semantic level.
+
+In short, syntax extraction gives us a way to “see the big picture” without having to manually dig through every line of code.
```

Comment:
I would like you to emphasize a bit more the fact that, representing a file through different methods, can lead to different representations, and consequently understanding by a computer that is comparing files using this information!

For example, consider that we could a) show all the file content to a computer or b) only the names of classes, variables and documentation, which capture semantic meaning tied to the code, or c) commits and issues discussion associated to the file (just like we are doing now), which capture others discussion surrounding the file. 

You could give a computer different representations (a, b, c) of a given file, and ask the computer to compare to another file (also represented in either a, b, c). Consider a computer would be having a different understanding of what two files are related depending on the representation of your choice:

It may say using criteria c) that the files are alike, because generally people are discussing similar features on its commits and issues, but if using a) the way the files were implemented being substantially different, they could be different.